sentinel-adapter/sentinel-api-gateway-adapter-common/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/common/param/ConfigurableRequestItemParser.java

```diff
@@ -0,0 +1,156 @@
+ *              return remoteAddress;
+ *          }
+ *      }
+ *      return null;
+ * });
+ * return new SentinelGatewayFilter(parser);
+ *
+ * @author icodening
+ * @date 2022.01.14
+ */
+public class ConfigurableRequestItemParser<T> implements RequestItemParser<T> {
+
+    private final List<Function<T, String>> pathExtractors = new ArrayList<>(2);
+
+    private final List<Function<T, String>> remoteAddressExtractors = new ArrayList<>(2);
+
+    private final List<BiFunction<T, String, String>> headerExtractors = new ArrayList<>(2);
+
+    private final List<BiFunction<T, String, String>> urlParamExtractors = new ArrayList<>(2);
+
+    private final List<BiFunction<T, String, String>> cookieValueExtractors = new ArrayList<>(2);
+
+    private final RequestItemParser<T> delegate;
+
+    public ConfigurableRequestItemParser(RequestItemParser<T> delegate) {
+        AssertUtil.notNull(delegate, "delegate can not be null");
+        this.delegate = delegate;
+    }
+
+    @Override
+    public String getPath(T request) {
+        for (Function<T, String> extractor : pathExtractors) {
+            String pathValue = extractor.apply(request);
+            if (StringUtil.isNotBlank(pathValue)) {
+                return pathValue;
+            }
+        }
+        return delegate.getPath(request);
+    }
+
+    @Override
+    public String getRemoteAddress(T request) {
+        for (Function<T, String> extractor : remoteAddressExtractors) {
+            String remoteAddress = extractor.apply(request);
+            if (StringUtil.isNotBlank(remoteAddress)) {
+                return remoteAddress;
+            }
+        }
+        return delegate.getRemoteAddress(request);
+    }
+
+    @Override
+    public String getHeader(T request, String key) {
+        for (BiFunction<T, String, String> extractor : headerExtractors) {
+            String headerValue = extractor.apply(request, key);
+            if (StringUtil.isNotBlank(headerValue)) {
+                return headerValue;
+            }
+        }
+        return delegate.getHeader(request, key);
+    }
+
+    @Override
+    public String getUrlParam(T request, String paramName) {
+        for (BiFunction<T, String, String> extractor : urlParamExtractors) {
+            String urlParam = extractor.apply(request, paramName);
+            if (StringUtil.isNotBlank(urlParam)) {
+                return urlParam;
+            }
+        }
+        return delegate.getUrlParam(request, paramName);
+    }
+
+    @Override
+    public String getCookieValue(T request, String cookieName) {
+        for (BiFunction<T, String, String> extractor : cookieValueExtractors) {
+            String cookie = extractor.apply(request, cookieName);
+            if (StringUtil.isNotBlank(cookie)) {
+                return cookie;
+            }
+        }
+        return delegate.getCookieValue(request, cookieName);
+    }
+
+    public void addPathExtractor(Function<T, String> extractor) {
```

Comment:
Maybe fluent-style API seems better? Just return `this` like a builder.

Reply:
OK, that's a good suggestion. I'll change it